lib/matplotlib/ticker.py

```diff
@@ -516,10 +516,10 @@ def _format_maybe_minus_and_locale(self, fmt, arg):
         Format *arg* with *fmt*, applying Unicode minus and locale if desired.
         """
         return self.fix_minus(
-                # Escape commas introduced by format_string but not those present
-                # from the beginning in fmt.
+                # Escape commas introduced by format_string if using math text, but not
+                # those present from the beginning in fmt.
```

Comment:
The "if using mathtext" clause should really apply everywhere (even though technically this doesn't change everything), i.e. I would write the comment as "If using mathtext, escape ... but not...", and the middle part of the code can probably be (equivalently)
```python
",".join(locale.format_string(part, (arg,)) for part in fmt.split(","))
if self._useMathText
else locale.format_string(fmt, (arg,))
```
(I think repeating `locale.format_string` is clearer than the no-op `replace(",", ",")`.)

The patch seems correct regardless, though.

Reply:
Sure, the `replace(",", ",")` is not that nice, but was the minimum change.

(I think the `replace(",", "{,}")` is missing from your suggestion, but I'll add that.)

Reply:
Not sure if I fully got your comment. I do not think that one should escape commas that are part of the users format? Only the ones introduced from the locale?

Anyway, I've updated it now to be a bit more clear with two locale.format_string.